src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/ElasticSearchDAOQuerySupport.scala

```diff
@@ -84,10 +80,13 @@ trait ElasticSearchDAOQuerySupport extends ElasticSearchDAOSupport {
     // if we are not collecting aggregation data (in the case of pagination), we can skip adding aggregations
     // if the search criteria contains elements from all of the aggregatable attributes, then we will be making
     // separate queries for each of them. so we can skip adding them in the main search query
-    if (criteria.fieldAggregations.nonEmpty && (criteria.fieldAggregations diff criteria.filters.keySet.toSeq).size != 0) {
-      // for the aggregations that are not part of the search criteria
-      // then the aggregation data for those fields will be accurate from the main search query so we add them here
-      addAggregationsToQuery(searchQuery, criteria.fieldAggregations.diff(criteria.filters.keySet.toSeq), criteria.maxAggregations)
+    if (criteria.fieldAggregations.nonEmpty) {
+      val fieldDiffs = criteria.fieldAggregations -- criteria.filters.keySet
+      if (fieldDiffs.size != 0) {
```

Comment:
I like using `fieldDiffs.nonEmpty` instead of `fieldDiffs.size != 0`, but that's trivial - certainly don't hold the PR for it